### PR TITLE
VP-1395: Implement order details feature

### PR DESCRIPTION
**Pull request by ilyawzrd**
Preview (with localization)
OrderDetails

### CODE REVIEW

File: client-app/src/pages/account/views/AccountOrders.vue [hunk 1]
```diff
@@ -102,5 +113,10 @@ export default class AccountOrders extends Vue{
     this.selectedOrder = items[0];
   }
 
+  openOrderDetails(orderId: string) {
```
Comment (trueboroda) [inline client-app/src/pages/account/views/AccountOrders.vue]: When opening order it should change selected order too.

File: client-app/src/pages/account/views/AccountOrderDetailsModal.vue [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+              {{ item.placedPrice.formattedAmount }}
+            </b-card-text>
+            <b-card-text>
+              <span class="font-weight-bold">{{ $t("account.orders.order-details.quantity") }}</span>
+              {{ item.quantity }}
+            </b-card-text>
+          </b-card-body>
+        </b-collapse>
+      </b-card>
+    </div>
```
Comment (trueboroda) [inline client-app/src/pages/account/views/AccountOrderDetailsModal.vue]: I guess we should extract the dummy component for OrderDetails that receive the order of type CustomerOrder on input as property.

File: client-app/src/pages/account/views/AccountOrders.vue [hunk 3]
```diff
@@ -14,7 +16,9 @@
                select-mode="single"
                @sort-changed="sortChanged"
                @row-selected="rowSelected">
-        <!-- A custom formatted column -->
+        <template v-slot:cell(number)="data">
+          <a @click="openOrderDetails(data.value)">{{ data.value }}</a>
```
Comment (trueboroda) [inline client-app/src/pages/account/views/AccountOrders.vue]: it does not look as link on view.

Comment (t13ka) [reply client-app/src/pages/account/views/AccountOrders.vue]: Try to remove highlighting after row clicking (current selected row). It looks strange 
image

**Comment by t13ka** [comment]
Not user-friendly popup behavior.
Look:  
1. I click to order number, then opens the empty popup:
image
and at the same time shows the preloader. I guess it is not correct.

2. when I open the popup window I get the old data from the previous opened order and then after getting the data from the server-side the data in the popup are refreshing with explicit blinking.

I suggest the following behavior:

1. user clicks on order number
2. ODT shows the preloader (gets data from server-side, placing the obtained data into placeholders, etc)
3. hides the preloader
4. opens the popup window with filled data (without any moment blinking)

User sees this and smiles because he always sees actual data in order details windows

File: client-app/src/pages/account/components/order-details/OrderDetailsTotals.vue [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+import { Prop } from "vue-property-decorator";
+import { CustomerOrder } from "@common/api/api-clients";
+
+@Component
+export default class OrderDetailsTotals extends Vue {
+  @Prop()
+  order!: CustomerOrder;
+
+  get shipping(): string | undefined {
+    return this.order.shippingTotal?.formattedAmount == "0,00 $" ? "Free" : this.order.shippingTotal?.formattedAmount;
```
Comment (t13ka) [inline client-app/src/pages/account/components/order-details/OrderDetailsTotals.vue]: Why do you using here explicit currency symbol? 
The question is: what will be if we use another currency in our site? We'll get "0,00 $"? It is strange...

I think we should get the current currency and put inside this interpolation string

File: client-app/src/pages/account/store/modules/orders-list/mutations.ts [hunk 5]
```diff
@@ -17,5 +21,15 @@ export const mutations: MutationTree<OrdersListState> = {
...
   [SET_ORDERS_LIST_CONFIG](state, payload: OrdersListConfig) {
     state.ordersList.listConfig = payload;
+  },
+  [SET_ORDER](state, payload: CustomerOrder) {
+    state.selectedOrder = payload;
+    state.loaded = true;
+    state.isLoading = false;
+    state.selectedOrderIsLoaded = true;
+  },
+  [CLEAR_ORDER_DETAIL](state) {
```
Comment (trueboroda) [inline client-app/src/pages/account/store/modules/orders-list/mutations.ts]: Rename to CLEAR_SELECDTED_ORDER. Action name does not match the state`s property name "selectedOrder".

**Pull request merged**